### PR TITLE
roachtest: rename DSC job compat test for 24.3; avoid pooling

### LINKED ISSUE #124442: roachtest: support declarative_schema_changer/job-compatibility-mixed-version test for 24.2
This issue tracks maintenance we have to do on the declarative_schema_changer/job-compatibility-mixed-version tests for a new CockroachDB version.

That is, with every major release, the `declarative_schema_changer/job-compatibility-mixed-version` test will need to get its set of dsc statements ran updated.

Jira issue: CRDB-38899

Epic CRDB-37763

### LINKED ISSUE #124791: roachtest: declarative_schema_changer/job-compatibility-mixed-version-V241-V242 failed
roachtest.declarative_schema_changer/job-compatibility-mixed-version-V241-V242 [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15444703?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/15444703?buildTab=artifacts#/declarative_schema_changer/job-compatibility-mixed-version-V241-V242) on master @ [d5f825ce04362a75781b75e19cf429f2ed5ed61b](https://github.com/cockroachdb/cockroach/commits/d5f825ce04362a75781b75e19cf429f2ed5ed61b):

```
(mixedversion.go:612).Run: mixed-version test failure while running step 19 (run "test step in mixed version"): pq: unique constraints without an index are not yet supported
test artifacts and logs in: /artifacts/declarative_schema_changer/job-compatibility-mixed-version-V241-V242/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-15444703/declarative-schema-changer-job-compatibility-mixed-version-v241/1716962523108/1716963268306)

</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*declarative_schema_changer/job-compatibility-mixed-version-V241-V242.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

Jira issue: CRDB-39074

### PR DESCRIPTION
**Pull request by rafiss**
See individual commits.

fixes https://github.com/cockroachdb/cockroach/issues/124442
fixes https://github.com/cockroachdb/cockroach/issues/124791
Release note: None

### CODE REVIEW

**Comment by rafiss** [comment]
started a trial run here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16123710?hideProblemsFromDependencies=false&hideTestsFromDependencies=false

File: pkg/cmd/roachtest/tests/mixed_version_job_compatibility_in_declarative_schema_changer.go [hunk 1]
```diff
@@ -205,7 +181,11 @@ CREATE VIEW IF NOT EXISTS testdb.testsc.v AS (SELECT i*2 FROM testdb.testsc.t);
 		// so that we don't fall back to legacy schema changer implicitly.
 		// Being explicit can help catch bugs that will otherwise be
 		// buried by the fallback.
+		// To make sure the session variables are applied correctly, we limit each
+		// connection pool to have at most 1 connection.
 		for _, node := range c.All() {
+			db := helper.Connect(node)
+			db.SetMaxOpenConns(1)
```
Comment (annrpom) [inline pkg/cmd/roachtest/tests/mixed_version_job_compatibility_in_declarative_schema_changer.go:188]: ah! nice -- ty

**Review by annrpom — APPROVED** [review]
lgtm, but can we change the name as well https://github.com/cockroachdb/cockroach/blob/36b4a9613c71ba0fd127ca5d3dae7013f465a353/pkg/cmd/roachtest/tests/mixed_version_job_compatibility_in_declarative_schema_changer.go#L34

**Comment by annrpom** [comment]
> lgtm, but can we change the name as well

ignore this

**Comment by rafiss** [comment]
bors retry

**Pull request merged**